Add Swagger specs for ARE[Azure Runtime Environment]

sushil490023 opened the pull request:
## ARM API Information (Control Plane)


<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ X] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. 
  **June 2023**

  4. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  **June 2023**
  5. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

Al supported SDKs will be updated for new API version

### Contribution checklist (MS Employees Only):
- [ X] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ X] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

raosuhas commented:
Can you please update the label

> Please ensure to respond feedbacks from the ARM API reviewer. When you are ready to continue the ARM API review, please remove `ARMChangesRequested`

Can you please update the label once PR is ready for ARM review ?

raosuhas commented:
> Hi, @sushil490023, For review efficiency consideration, when creating a new api version, it is required to place API specs of the base version in the first commit, and push new version updates into successive commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Or you could onboard [API spec pipeline](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/606/Enable-API-Spec-Pipeline)

Please fix this comment as well

Commit:
First Commit of Stable version

zedy-wj commented:
@sushil490023 - Please fix CI errors according to the [comments](https://github.com/Azure/azure-rest-api-specs/pull/23988#issuecomment-1549004353).

sushil490023 commented:
Hi @zedy-wj 

Can you help/review the PR 
1. Avocado error: We are depreciating Module swagger with new Package swagger.
2. Lint Diff error 
a. **ArmResourcePropertiesBag** : We need "**Type**" property in Runbook model.
b. **PutRequestResponseSchemeArm** : We cant provide lastmodified in request model of Put call as it wont be present..
3. **Swagger SemanticValidation**  : Not sure in which example its failing.

Can you help with errors .

raych1 commented:
> Hi @zedy-wj
> 
> Can you help/review the PR
> 
> 1. Avocado error: We are depreciating Module swagger with new Package swagger.
> 2. Lint Diff error
>    a. **ArmResourcePropertiesBag** : We need "**Type**" property in Runbook model.
>    b. **PutRequestResponseSchemeArm** : We cant provide lastmodified in request model of Put call as it wont be present..
> 3. **Swagger SemanticValidation**  : Not sure in which example its failing.
> 
> Can you help with errors .

I approved avocado errors based on above justification. 

For LintDiff errors, please reach out ARM on-call person to approve.

For semanticValidation errors, the errors are on swagger, please click on the error link for the details.

sushil490023 commented:
@rkmanda Can you check why the Validation tests are stuck on queue stage, all required tests have been passed last time. i Just squashed multiple commits into one.

Also i have requested for Breaking changes review. Can you please also review it.

sushil490023 commented:
@rkmanda Can you please Review PR.

mentat9 commented:
>         "name": {

Don't duplicate top-level property names within the property bag. Consider runtimeEnvironmentName or similar.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/job.json:852 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "runtimeEnvironment": {

Why is this a string in the create request, but an object in the collection response? Can you make them consistent?

The ARM RPC actually requires that GET, PUT, and collection GET all return the same payload and (although not a requirement) a very consistent defacto standard is that the PUT request payload also matches. Even though this API doesn't follow the RPC, getting closer to it as it gets updated is desirable.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/job.json:980 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>             "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"

Consider moving to v5 common types for new APIs. Comment applies throughout the updates in this PR.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:56 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "name": {

Recommend replacing name, location, and tags with a reference to the TrackedResource definition from v5 common types.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:509 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>           "enum": [

Canceled (spelled like that) is a a required value.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:549 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "creationTime": {

This and lastModifiedTime belong in the systemData top-level object (see TrackedResource definition in v5 common types).

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:580 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "name": {

Recommend replacing name, location, and tags with a reference to the TrackedResource definition from v5 common types.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:615 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>           "$ref": "../../common/v1/definitions.json#/definitions/TrackedResource"

Recommend using v5 common types

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:648 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>   "paths": {

You need to add a collection GET path.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:38 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "runtimeEnvironment": {

Why move to unrestricted string? Client APIs with enums are better for customers.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runbook.json:1180 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "runbookType": {

Why move to unrestricted string for this property?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runbook.json:1323 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "runtimeEnvironmentName": {

Why is this property named runtimeEnvironmentName here and just runtimeEnvironment in other definitions?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runbook.json:1569 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "runbookType": {

Why not enum?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runbook.json:1658 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "ActivityListResult": {

I don't see where this is used anywhere. You should remove it if it's not used.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:423 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "Activity": {

This appears to only be used in an unused definition, so should also be removed.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:403 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "ActivityProperties": {

This also appears to be unused (except in unused definitions).

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:357 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "ActivityParameterSet": {

Used only by unused definitions?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:315 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "ActivityParameter": {

Used only by unused definitions.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:264 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "ActivityParameterValidationSet": {

Used only by unused definitions.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:333 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "ActivityOutputType": {

Used only by unused definitions.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:343 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "location": {

Recommend using the TrackedResource definition in v5 common types. At minimum, why not use your own TrackedResource definition?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runtimeEnvironment.json:304 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "creationTime": {

creationTime and lastModifiedTime should be in the top-level systemData object. This is already defined for you in TrackedResource (or Resource) definition from v5 common types.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runtimeEnvironment.json:377 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>     "DefaultPackages": {

The additionalProperties pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runtimeEnvironment.json:398 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented:
>         "tags": {

You can use TrackedResource from v5 common types for this.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runtimeEnvironment.json:355 in a571cd1. [](commit_id = a571cd16481772faf625e99de4a8ded3363151da, deletion_comment = False)

mentat9 commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package-lock.json:
This file appears to be leftover NPM junk. Not part of a spec, so should be removed.

sushil490023 commented:
> > ```
> >     "name": {
> > ```
> 
> Don't duplicate top-level property names within the property bag. Consider runtimeEnvironmentName or similar.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/job.json:852 in [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da). [](commit_id = [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da), deletion_comment = False)

replaced with runtimeEnvironmentName

sushil490023 commented:
> Canceled



> > ```
> >     "name": {
> > ```
> 
> Recommend replacing name, location, and tags with a reference to the TrackedResource definition from v5 common types.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:615 in [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da). [](commit_id = [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da), deletion_comment = False)

Package is not tracked resource here

sushil490023 commented:
> Activity



> > ```
> >     "runtimeEnvironment": {
> > ```
> 
> Why is this a string in the create request, but an object in the collection response? Can you make them consistent?
> 
> The ARM RPC actually requires that GET, PUT, and collection GET all return the same payload and (although not a requirement) a very consistent defacto standard is that the PUT request payload also matches. Even though this API doesn't follow the RPC, getting closer to it as it gets updated is desirable.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/job.json:980 in [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da). [](commit_id = [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da), deletion_comment = False)

Please don't confuse it with **runtimeEnvironment Name** which is passed along with request and response. Its an entity, we did this change intentionally as going forward in future we want to return more information with payload instead of Just Name of environment.

sushil490023 commented:
> > ```
> >         "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
> > ```
> 
> Consider moving to v5 common types for new APIs. Comment applies throughout the updates in this PR.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:56 in [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da). [](commit_id = [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da), deletion_comment = False)

Updated all mentioned references with V5 in all refences mentioned in PR comments

sushil490023 commented:
> > ```
> >       "enum": [
> > ```
> 
> Canceled (spelled like that) is a a required value.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:549 in [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da). [](commit_id = [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da), deletion_comment = False)

Updated

sushil490023 commented:
> > ```
> >     "runtimeEnvironment": {
> > ```
> 
> Why move to unrestricted string? Client APIs with enums are better for customers.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runbook.json:1180 in [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da). [](commit_id = [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da), deletion_comment = False)

These are name which are provided by customer as part of input and are dynamic

sushil490023 commented:
> > ```
> >     "runbookType": {
> > ```
> 
> Why move to unrestricted string for this property?
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runbook.json:1323 in [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da). [](commit_id = [a571cd1](https://github.com/Azure/azure-rest-api-specs/commit/a571cd16481772faf625e99de4a8ded3363151da), deletion_comment = False)

This was done intentionally as going forward we want to add support for many upcoming Runbook type without causing the breaking changes.

sushil490023 commented:
@mentat9 First thing Thank you so much for reviewing in Depth. I have addressed your comments and provided justification wherever required let me know if any further changes are required.

mentat9 commented:
@sushil490023 - You need to remove the ARMChangesRequested label to get your PR back on ARM review radar. IIRC, you might not have had permission for that, so I've done it for you. Current on-call (@raosuhas) will resume your ARM review.

Be sure to follow up with @rkmanda (Roopesh Manda) to get your permissions fixed: you cannot work in this repo unless you can set labels.

sushil490023 commented:
@raosuhas  can you please review The PR ?

sushil490023 commented:
@raosuhas @mentat9 @zedy-wj Can someone please review the PR

visingla-ms commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/account.json:
```diff
@@ -0,0 +1,964 @@
+  "swagger": "2.0",
+  "info": {
+    "title": "AutomationManagement",
+    "version": "2023-05-15-preview",
```
Please check the breaking changes comment: [https://github.com/Azure/azure-rest-api-specs/pull/23988#issuecomment-1634101907](https://github.com/Azure/azure-rest-api-specs/pull/23988#issuecomment-1634101907). ARM will review your PR changes once breaking changes are resolved.

Commit:
Add Swagger file for ARE

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/account.json:
```diff
@@ -0,0 +1,1085 @@
+        "runtime": {
+          "type": "string",
+          "description": "Gets or sets runtime of the runbook.",
+          "x-nullable": true
```
>   "x-nullable": true [](http://example.com/codeflow?start=7&length=21)

As a general rule, we don't recommend for azure APIs, using x-nullable, which results in explicit 'null' in the response. Just omitting the property is preferred.

TimLovellSmith commented:
>   "paths": {

(/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/runtimeEnvironments/{runtimeEnvironmentName}/packages/ in this case)

---
In reply to: [1626383025](https://github.com/Azure/azure-rest-api-specs/pull/23988#issuecomment-1626383025) [](http://example.com/codeflow?ancestors=1626383025)

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:38 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/common/v5/types.json:
```diff
@@ -0,0 +1,721 @@
+    "title": "Common types"
+  },
+  "paths": {},
+  "definitions": {
```
Factoringwise, I'm wondering why add this instead of just referencing the usual shared common-types v5 definitions directly?
Are these just an exact copy or are there interesting differences?

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/createOrUpdateConnection.json:
```diff
@@ -0,0 +1,62 @@
+          },
+          "connectionType": {
+            "name": "Azure"
+          }
```
curious - why doe it look like this instead of just e.g.

   "connectionType": "Azure"
?

Jassingh819 replied:
This ConnectionType is further object under Connection-
specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/createOrUpdateConnectionType.json

TimLovellSmith commented:
>           "Default": false,

why is 'Default' capitalized?
And what does default = false actually mean?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/package/getPackage.json:20 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>             "code": null,

Its generally preferred to omit properties instead of returning explicit null.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/package/getPackage.json:24 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>         "description": "Delete the Package by name.",

Docs (and descriptions) are supposed to be plain english - and package is just an ordinary noun, not a proper noun or acronym, so it shouldn't be capitalized.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:45 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>         "description": "Create or Update the package identified by package name.",

Docs (and descriptions) are supposed to be plain english - and update is just an ordinary verb, not a proper noun or acronym, so it shouldn't be capitalized

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:151 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>             "Created",

Created shouldn't be used... unless its a *temporary* state, after which it will finally go to Succeeded, Canceled, or Failed.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:428 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>             "Created",

(Even then it may be a confusing name to the api consumers)

---
In reply to: [1680874357](https://github.com/Azure/azure-rest-api-specs/pull/23988#issuecomment-1680874357) [](http://example.com/codeflow?ancestors=1680874357)

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:428 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>         "operationId": "Package_Get",

should be Packages_List or Packages_ListByRuntimeEnvironment etc

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:282 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>           "description": "Gets or sets the error code."

you allow setting it?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:335 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>         "Default": {

camelCase property names

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:411 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>         "provisioningState": {

provisioningState should be readonly

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:424 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
>           "description": "list all the Packages of RuntimeEnvironment."

This description should describe the 'value' property of the package list result. Currently it does not, it describes the list operation.

>   "description": "list all the Packages of RuntimeEnvironment."

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:513 in ea69785. [](commit_id = ea69785dc318a05d8ae62790e3237bdb0a44a0e8, deletion_comment = False)

TimLovellSmith commented:
If you could please confirm - is first commit the base commit that is just a copy of the old api version and updates the api version string without other changes? (Its easier not to get asked this question if you just use open api hub...)

sushil490023 commented:
> :



> > ```
> >       "Default": false,
> > ```
> 
> why is 'Default' capitalized? And what does default = false actually mean?
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/package/getPackage.json:20 in [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8). [](commit_id = [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8), deletion_comment = False)

corrected, default here means its package which customer has not created/uploaded  but something which is provided by Automation account by default.

sushil490023 commented:
> description



> > ```
> >       "description": "list all the Packages of RuntimeEnvironment."
> > ```
> 
> doc style nit instead of the programmerese - 'RuntimeEnvironment' try to talk like a real person and say e.g. 'the runtime environment'. Conversational style.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:513 in [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8). [](commit_id = [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8), deletion_comment = False)

Updated

sushil490023 commented:
> > ```
> >     "description": "Delete the Package by name.",
> > ```
> 
> Docs (and descriptions) are supposed to be plain english - and package is just an ordinary noun, not a proper noun or acronym, so it shouldn't be capitalized.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:45 in [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8). [](commit_id = [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8), deletion_comment = False)

updated

sushil490023 commented:
> Packages_ListByRuntimeEnvironment



> If you could please confirm - is first commit the base commit that is just a copy of the old api version and updates the api version string without other changes? (Its easier not to get asked this question if you just use open api hub...)

Yes i can confirm first commit is just copy of the old api version and updates the api version string without other changes. As required.

sushil490023 commented:
> > ```
> >         "Created",
> > ```
> 
> Created shouldn't be used... unless its a _temporary_ state, after which it will finally go to Succeeded, Canceled, or Failed.
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/package.json:428 in [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8). [](commit_id = [ea69785](https://github.com/Azure/azure-rest-api-specs/commit/ea69785dc318a05d8ae62790e3237bdb0a44a0e8), deletion_comment = False)

Yes its a temporary state for aysnc call response which will finally succeed. or fail.

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/account.json:
```diff
@@ -0,0 +1,1084 @@
+        "runtimeEnvironment": {
+          "type": "string",
+          "description": "Environment of the runbook.",
+          "x-nullable": true
```
>"x-nullable": true [](http://example.com/codeflow?start=10&length=18)

strings don't usually need x-nullable, so I'm wondering if there's some special reason again.

if you want to distinguish strings that are always present from ones which are not, 'required' could be the way to do that.

TimLovellSmith replied:
but if its already shipped its probably a breaking change so never mind..

sushil490023 replied:
Got its, Yes its already shipped

TimLovellSmith commented:
>             "type": "Microsoft.Automation/AutomationAccounts/RuntimeEnvironment",

type names should be consistently camel case: 'runtimeEnvironment

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/runtimeEnvironment/listRuntimeEnvironmentsByAutomationAccount.json:37 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented:
>             "type": "Microsoft.Automation/AutomationAccounts/RuntimeEnvironment",

Also you have to keep the plurality: runtimeEnvironments

---
In reply to: [1684063724](https://github.com/Azure/azure-rest-api-specs/pull/23988#issuecomment-1684063724) [](http://example.com/codeflow?ancestors=1684063724)

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/runtimeEnvironment/listRuntimeEnvironmentsByAutomationAccount.json:37 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/stopJob.json:
```diff
@@ -0,0 +1,12 @@
+    "api-version": "2023-05-15-preview"
+  },
+  "responses": {
+    "200": {}
```
>"200": {} [](http://example.com/codeflow?start=4&length=9)

200 should have a body. when there's no body it can be 204

sushil490023 replied:
There has been no changes in Job Swagger and this is already shipped. Also from API also we only return 200 with empty body for stopJob, resumeJob, suspendJob operations changing it to 204 would result in breaking change.

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/resumeJob.json:
```diff
@@ -0,0 +1,12 @@
+    "api-version": "2023-05-15-preview"
+  },
+  "responses": {
+    "200": {}
```
>  "200": {} [](http://example.com/codeflow?start=2&length=11)

200 should have a body. when there's no body it can be 204

sushil490023 replied:
There has been no changes in Job Swagger and this is already shipped. Also from API also we only return 200 with empty body for stopJob, resumeJob, suspendJob operations changing it to 204 would result in breaking change.

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/suspendJob.json:
```diff
@@ -0,0 +1,12 @@
+    "api-version": "2023-05-15-preview"
+  },
+  "responses": {
+    "200": {}
```
>  "200": {} [](http://example.com/codeflow?start=2&length=11)

200 should have a body. when there's no body it can be 204

sushil490023 replied:
There has been no changes in Job Swagger and this is already shipped. Also from API also we only return 200 with empty body for stopJob, resumeJob, suspendJob operations changing it to 204 would result in breaking change.

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/listJobsByAutomationAccount.json:
```diff
@@ -0,0 +1,59 @@
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/51766542-3ed7-4a72-a187-0c8ab644ddab/resourceGroups/mygroup/providers/Microsoft.Automation/automationAccounts/ContoseAutomationAccount/jobs/job1",
```
>Contose [](http://example.com/codeflow?start=144&length=7)

nit: typo Contose -> Contoso

sushil490023 replied:
Updated

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/getJobStream.json:
```diff
@@ -0,0 +1,26 @@
+          "time": "2018-02-07T02:48:11.0693884+00:00",
+          "streamType": "Output",
+          "streamText": "",
+          "value": {}
```
>   "value": {} [](http://example.com/codeflow?start=7&length=14)

what is value here for if its empty?

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/getJobStream.json:
```diff
@@ -0,0 +1,26 @@
+        "id": "/subscriptions/51766542-3ed7-4a72-a187-0c8ab644ddab/resourcegroups/mygroup/providers/Microsoft.Automation/automationAccounts/ContoseAutomationAccount/jobs/jobName/streams/851b2101-686f-40e2-8a4b-5b8df08afbd1_00636535684910693884_00000000000000000001",
+        "properties": {
+          "jobStreamId": "851b2101-686f-40e2-8a4b-5b8df08afbd1:00636535684910693884:00000000000000000001",
+          "summary": null,
```
>"summary": null, [](http://example.com/codeflow?start=10&length=16)

avoid string nullable properties

>Set "x-nullable": true on a schema to indicate that a null is a legal value. By default, a null value should be disallowed when forming a request and rejected during payload deserialization.

For arrays, sending/receiving a null array entry is not supported and should result in an error.

sushil490023 replied:
Marked "x-nullable": true for summary

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/getJobOutput.json:
```diff
@@ -0,0 +1,17 @@
+      "headers": {
+        "Content-Type": "text/plain"
+      },
+      "body": "Wednesday, February 7, 2018 3:47:17 PM"
```
>body [](http://example.com/codeflow?start=7&length=4)

Having a string response is not easy to extend in the future without breaking clients. I'd suggest you change it to a JSON response unless there's a strong reason not to

TimLovellSmith commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/job/getJobRunbookContent.json:
```diff
@@ -0,0 +1,17 @@
+      "headers": {
+        "Content-Type": "text/powershell"
+      },
+      "body": "get-date"
```
>    "body": "get-date" [](http://example.com/codeflow?start=2&length=22)

Having a string response is not easy to extend in the future without breaking clients. I'd suggest you change it to a JSON response unless there's a good reason not to

(And I can't help feeling curious why this would be the result...)

sushil490023 replied:
This is existing changes which are already shipped , we would consider this for any new changes.

TimLovellSmith commented:
>         "etag": null,

I don't think etag should really ever be null in your responses unless maybe we're talking about a partially created resource where provisioningState isn't finished?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/updateAutomationAccount.json:26 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented:
>             "capacity": null

Ideally family and capacity shouldn't be explicit nulls, they should just be omitted.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/updateAutomationAccount.json:31 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented:
>         "graphRunbookJson": "<GraphRunbookJSON>"

Is it a going to be a string quoted JSON? Why not a json response?

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/serializeGraphRunbookContent.json:20 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented:
>     "200": {}

HTTP idiomaticness - 200 response should have a body otherwise it can be 204

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/moveHybridRunbookWorker.json:14 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented:
>             "description": "Parameters supplied to the create  RuntimeEnvironment."

doc style: Don't call it 'RuntimeEnvironment' the typename... call it 'the runtime environment' or whatever your docs would call it.

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runtimeEnvironment.json:123 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented:
>             "description": "OK"

200 must have a schema

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runtimeEnvironment.json:186 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

TimLovellSmith commented:
>         "systemData": {

this is incorrect - systemData is supposed to be in the top level envelope!

---
Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/runtimeEnvironment.json:366 in 554b591. [](commit_id = 554b5917e57211de88e1e8f9cd10c128e5b404e0, deletion_comment = False)

sushil490023 commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/account.json:
```diff
@@ -0,0 +1,1084 @@
+          "format": "date-time",
+          "readOnly": true,
+          "description": "Gets the endTime of the statistic.",
+          "x-nullable": true
```
Yes its already shipped

sushil490023 commented:
> > ```
> >         "type": "Microsoft.Automation/AutomationAccounts/RuntimeEnvironment",
> > ```
> 
> type names should be consistently camel case: 'runtimeEnvironment [ARMShouldGuidelineViolation]
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/runtimeEnvironment/listRuntimeEnvironmentsByAutomationAccount.json:37 in [554b591](https://github.com/Azure/azure-rest-api-specs/commit/554b5917e57211de88e1e8f9cd10c128e5b404e0). [](commit_id = [554b591](https://github.com/Azure/azure-rest-api-specs/commit/554b5917e57211de88e1e8f9cd10c128e5b404e0), deletion_comment = False)

Updated

sushil490023 commented:
> > ```
> >     "type": "Microsoft.Automation/AutomationAccounts/Runbooks",
> > ```
> 
> should capitalize the same way as in resource ides
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/runbook/updateRunbook.json:22 in [554b591](https://github.com/Azure/azure-rest-api-specs/commit/554b5917e57211de88e1e8f9cd10c128e5b404e0). [](commit_id = [554b591](https://github.com/Azure/azure-rest-api-specs/commit/554b5917e57211de88e1e8f9cd10c128e5b404e0), deletion_comment = False)

its already shipped and there has been no changes in Jobs swagger, changing here would result in breaking changes, for upcoming new resources we will keep this in mind.

sushil490023 commented:
> > ```
> > "200": {}
> > ```
> 
> HTTP idiomaticness - 200 response should have a body otherwise it can be 204
> 
> Refers to: specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/examples/moveHybridRunbookWorker.json:14 in [554b591](https://github.com/Azure/azure-rest-api-specs/commit/554b5917e57211de88e1e8f9cd10c128e5b404e0). [](commit_id = [554b591](https://github.com/Azure/azure-rest-api-specs/commit/554b5917e57211de88e1e8f9cd10c128e5b404e0), deletion_comment = False)

There has been no changes in HYbridWorkerSwagger and this is already shipped. Also from API also we only return 200 with empty body for stopJob, resumeJob, suspendJob operations changing it to 204 would result in breaking change.

sushil490023 commented:
> listVariables_Next100.

Updated

sushil490023 commented:
> runtimeEnvironment

Updated

Alancere commented:
please fix azure-sdk-for-go ci failure

sushil490023 commented:
@Alancere Hi we don't support Azure GO SDK for azure automation

sushil490023 commented:
@Alancere Can someone from Azure GO SDK team help with the error ?

Not sure how to fix error "NING: Unable to find version for /mnt/vss/_work/1/s/azure-sdk-for-go/src/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/automation/armautomation"

Alancere commented:
/azp run

Alancere commented:
/azp run

Alancere committed:
rename name to packageName

Alancere commented:
Hi @sushil490023 , please fix `Duplicate Schema named` error

Alancere committed:
fix Package operation examples

Commit:
Fixed Avacado errors

Commit:
Fixed duplicate schema

sushil490023 commented:
/azp run

Commit:
Fix Duplicate schema erors.

Commit:
Fix Duplicate schema erors.

sushil490023 commented:
Hi 
Can someone help with following validation failing 
[Add Swagger specs for ARE[Azure Runtime Environment] by sushil490023 · Pull Request #23988 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/23988/checks?check_run_id=16471077811)


Its older spec which has just been moved to new API version

Commit:
Fix Duplicate Schema error

Commit:
Fix Errors

sushil490023 commented:
> Hi Can someone help with following validation failing ? @TimLovellSmith @raosuhas @visingla-ms @rkmanda [Add Swagger specs for ARE[Azure Runtime Environment] by sushil490023 · Pull Request #23988 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/23988/checks?check_run_id=16471077811)
> 
> Its older spec which has just been moved to new API version

In Validation, Error Message is not sufficient to Fix it **"Expected type but found type object"** What type its expecting, 

Also not able to understand for below error
<img width="608" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/25101734/996c8bb6-37be-4b19-b639-764fdd861e07">

It Swagger its defined as Addtional property with object type as its can be String, boolean anything.

<img width="368" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/25101734/bfedd988-1ad7-4986-9f83-7c0546ee1da6">

We cant fix it to a particular type

raosuhas commented:
> In Validation, Error Message is not sufficient to Fix it **"Expected type but found type object"** What type its expecting,

The response for a GET needs to be typed to a specific object , you cannot just return a 200 response without defining the type that is returned. That is the issue with the GET 

For the ones that talk about additional property , it is a similar issue , you need to specify a specific type. Additional properties are hard for clients to use, don't produce good documentation, and don't work with core ARM features like Azure Policy and ARG. They aren't actually strongly typed, because discovery of the properties can only be done at runtime. This is why they are flagged.

Commit:
Testing

Commit:
Fix errors

pcsin commented on specification/automation/resource-manager/Microsoft.Automation/common/v5/definitions.json:
```diff
@@ -0,0 +1,118 @@
+  },
+  "paths": {},
+  "definitions": {
+    "PrivateEndpointConnection": {
```
Pls check to refer these from common types. I see `privateEndpoint` related object types are similar to the ones available in the common-types here - `specification/common-types/resource-management/v5/privatelinks.json`

pcsin commented on specification/automation/resource-manager/Microsoft.Automation/common/v5/definitions.json:
```diff
@@ -0,0 +1,118 @@
+        }
+      ]
+    },
+    "PrivateEndpointConnectionProperties": {
```
Pls consider using these properties from `common-types`

mahakjain314 commented:
We got the ARM signoff last month but we were facing some Go SDK errors. To resolve those, we pushed some more changes mainly fixing few contracts and validations.
![image](https://github.com/Azure/azure-rest-api-specs/assets/50352900/50605d02-dc6f-438d-a9df-1e38866787b7)

mahakjain314 commented:
We are facing two blockers currently in merging this PR which are not fixable because **they are inherited from older stable APIs in which we have done no changes**. Any change to those would be breaking change.

![image](https://github.com/Azure/azure-rest-api-specs/assets/50352900/0d46b66c-0211-4a1e-8368-2be26b9c30fc)
This is GUID by design and cannot be changed.

![image](https://github.com/Azure/azure-rest-api-specs/assets/50352900/3eee74f6-fe52-4b5d-bd9f-22cc952c357c)
This is an existing resource for many years and changing the response model to any specific contract instead of generic object would be a breaking change.

Need some assistance in suppressing these two validations to merge this PR.

mahakjain314 commented:
> > In Validation, Error Message is not sufficient to Fix it **"Expected type but found type object"** What type its expecting,
> 
> The response for a GET needs to be typed to a specific object , you cannot just return a 200 response without defining the type that is returned. That is the issue with the GET
> 
> For the ones that talk about additional property , it is a similar issue , you need to specify a specific type. Additional properties are hard for clients to use, don't produce good documentation, and don't work with core ARM features like Azure Policy and ARG. They aren't actually strongly typed, because discovery of the properties can only be done at runtime. This is why they are flagged.

Understood the intent and we tried to address these in the APIs, but it would lead to breaking change for customers. These are very old APIs which are not actively being worked upon but are inherited from older stable APIs. Thus, we cannot make modifications to these.

mahakjain314 committed:
Removed uuid format for PrincipalId in Identity properties

mahakjain314 committed:
Fix example for ModelValidation Failure

mahakjain314 committed:
Re-run Go SDK

pratikjoshi14 commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/deletedAutomationAccount.json:
```diff
@@ -0,0 +1,144 @@
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the automation account properties."
+        },
+        "id": {
```
>id [](http://example.com/codeflow?start=9&length=2)

You can use below common type for tracked resource:

      "allOf": [
        {
          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
        }

mahakjain314 replied:
We have not modified this resource. It came as part of the last stable version.

pratikjoshi14 commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/deletedAutomationAccount.json:
```diff
@@ -0,0 +1,144 @@
+    "DeletedAutomationAccountProperties": {
+      "type": "object",
+      "properties": {
+        "automationAccountResourceId": {
```
>automationAccountResourceId [](http://example.com/codeflow?start=9&length=27)

Is it an azure resource ID? If so, please add "format": "arm-id" and "x-ms-arm-id-details".

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details)

mahakjain314 replied:
We have not modified this resource. It came as part of the last stable version.

pratikjoshi14 commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/certificate.json:
```diff
@@ -0,0 +1,452 @@
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CertificateCreateOrUpdateParameters"
```
>CertificateCreateOrUpdateParameters [](http://example.com/codeflow?start=37&length=35)

Any reason why PUT object and GET object are not the same?

mahakjain314 replied:
We have not modified this resource. It came as part of the last stable version.

pratikjoshi14 commented:
Since multiple reviews have happened on the PR, approved from AzCoreIDC side. Please get a final approval from ARM team.

rkmanda commented on specification/automation/resource-manager/Microsoft.Automation/preview/2023-05-15-preview/account.json:
```diff
@@ -0,0 +1,1084 @@
+        },
+        "runbookType": {
+          "type": "string",
+          "description": "Type of the runbook."
```
enum?

rkmanda reviewed (approved):
:shipit:
